myia/parser.py

```diff
@@ -313,9 +313,6 @@ def process_while(self, block: 'Block', node: ast.While) -> 'Block':
         header_block = Block(self)
         body_block = Block(self)
         after_block = Block(self)
-        header_block.preds.extend([block, body_block])
```

Comment:
@breuleux This is were things were added twice, the graph looks okay now to me, haven't tested it with the VM.